### PR TITLE
[3.1] Fixes Producer Plugin Schema Fixes and Better Alignment with OpenAPI spec

### PR DESCRIPTION
**Pull request by ericpassmore**
### Bug Fixes
- spelling fixes
- added `subjective_cpu_leeway_us` to runtime options 
- added `more` option array returned from `get_account_ram_corrections`
- added return boolean to `paused`
- correct `201` return codes 
- `create_snapshot` has return schema previous was *Returns Nothing*
- corrected description for `get_whitelist_blacklist` `set_whitelist_blacklist`

### Schema Alignment
- consolidates to `components` removed hanging `component` section
- moved common schemas to `components` section 
- removed empty `requestBody` 
- added `400` return codes as required by `redocly` linter
- Added version tags to clarify protocol version 

**NOTE** issue #547 will be closed after patching and updating openapi docs in 3.2 and main branches 
Related PR #572 - 3.2 release

### CODE REVIEW

File: plugins/producer_api_plugin/producer.swagger.yaml [hunk 1]
```diff
@@ -22,477 +23,607 @@ servers:
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+
   /producer/get_runtime_options:
     post:
       summary: get_runtime_options
-      description: Retreives run time options for producer node
+      description: Retrieves runtime options for producer node. `incoming_defer_ratio` is a floating point number. The other return values are integers.
```
Comment (lparisc) [inline plugins/producer_api_plugin/producer.swagger.yaml:89]: Probably not a good idea to mention data types on description. It creates another source of truth dependency. For instance, this statement might be incorrect in the future: The other return values are integers.

File: plugins/producer_api_plugin/producer.swagger.yaml [hunk 2]
```diff
@@ -22,477 +23,607 @@ servers:
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+
   /producer/update_runtime_options:
     post:
       summary: update_runtime_options
-      description: Update run time options for producer node
+      description: Update runtime options for producer node. May post any of the runtime options in combination or alone. `incoming_defer_ratio` is a floating point number. The other values are integers.
```
Comment (lparisc) [inline plugins/producer_api_plugin/producer.swagger.yaml:108]: Same as above.

File: plugins/producer_api_plugin/producer.swagger.yaml [hunk 3]
```diff
@@ -22,477 +23,607 @@ servers:
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
 
   /producer/get_whitelist_blacklist:
     post:
       summary: get_whitelist_blacklist
-      description: Retreives the white list and black list for producer node
+      description: Retrieves the whitelist and blacklist for producer node. A JSON object containing whitelist and blacklist information. `actor_whitelist`, `actor_blacklist`, `contract_whitelist`, `contract_blacklist` are represented by an array of Names. `action_blacklist` is an array of tuples consisting of Name and Action. `key_blacklist` is an array of Public Keys. Name is a string representing a NamePrivileged or NameBasic or NameBid or NameCatchAll
```
Comment (lparisc) [inline plugins/producer_api_plugin/producer.swagger.yaml:217]: Same as above.

File: plugins/producer_api_plugin/producer.swagger.yaml [hunk 4]
```diff
@@ -22,477 +23,607 @@ servers:
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
 
   /producer/set_whitelist_blacklist:
     post:
       summary: set_whitelist_blacklist
-      description: Sets the white list and black list for producer node
+      description: Defines the whitelist and blacklist for a producer node. Takes a JSON object containing whitelist and blacklist information. At least one of actor_whitelist, actor_blacklist, contract_whitelist, contract_blacklist, action_blacklist, and key_blacklist is required. `actor_whitelist`, `actor_blacklist`, `contract_whitelist`, `contract_blacklist` take an array of Names. `action_blacklist` is an array of tuples consisting of Name and Action. `key_blacklist` is an array of Public Keys. Name is a string represeting a NamePrivileged or NameBasic or NameBid or NameCatchAll
```
Comment (lparisc) [inline plugins/producer_api_plugin/producer.swagger.yaml]: Same as above.

File: plugins/producer_api_plugin/producer.swagger.yaml [hunk 5]
```diff
@@ -22,477 +23,607 @@ servers:
...
+          type: integer
+          description: in micro seconds
+          example: 10
+        incoming_defer_ratio:
+          type: string
+          description: Incoming defer ratio, parsed to double
+          example: "1.00000000000000000"
+        greylist_limit:
+          type: integer
+          description: unsigned int, limit on number of Names supported by greylist
```
Comment (lparisc) [inline plugins/producer_api_plugin/producer.swagger.yaml:628]: Same as above. Not necessary to mention explicit data type (unsigned int). Maybe "numerical limit on number of...".

File: plugins/producer_api_plugin/producer.swagger.yaml [hunk 6]
```diff
@@ -22,477 +23,607 @@ servers:
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
 
   /producer/set_whitelist_blacklist:
     post:
       summary: set_whitelist_blacklist
-      description: Sets the white list and black list for producer node
+      description: Defines the whitelist and blacklist for a producer node. Takes a JSON object containing whitelist and blacklist information. At least one of actor_whitelist, actor_blacklist, contract_whitelist, contract_blacklist, action_blacklist, and key_blacklist is required. `actor_whitelist`, `actor_blacklist`, `contract_whitelist`, `contract_blacklist` take an array of Names. `action_blacklist` is an array of tuples consisting of Name and Action. `key_blacklist` is an array of Public Keys. Name is a string representing a NamePrivileged or NameBasic or NameBid or NameCatchAll
```
Comment (lparisc) [inline plugins/producer_api_plugin/producer.swagger.yaml:266]: Same as above.

**Review by lparisc — APPROVED** [review]
Just some minor suggestions, pushed some minor edits, and approved.

File: plugins/producer_api_plugin/producer.swagger.yaml [hunk 7]
```diff
@@ -22,477 +23,607 @@ servers:
...
+                          example: true
+                        preactivation_required:
+                          type: boolean
+                          example: true
+                        earliest_allowed_activation_time:
+                          type: string
+                          example: "1970-01-01T00:00:00.000"
+                        description_digest:
+                          $ref: "https://docs.eosnetwork.com/openapi/v2.0/Sha256.yaml"
+                        dependancies:
```
Comment (iamveritas) [inline plugins/producer_api_plugin/producer.swagger.yaml:455]: isn't this `dependencies`?

File: plugins/producer_api_plugin/producer.swagger.yaml [hunk 8]
```diff
@@ -22,477 +23,607 @@ servers:
...
+                    type: string
+                    description: file where error was thrown
+                    example: http_plugin.hpp
+                  line_number:
+                    type: integer
+                    description: line number in file where error was thrown
+                    example: 246
+                  method:
+                    type: string
+                    description: function executed when error occured
```
Comment (iamveritas) [inline plugins/producer_api_plugin/producer.swagger.yaml:579]: typo: occured -> occurred

**Review by iamveritas — COMMENTED** [review]
found a couple of typos and I also agree with Luis' comments.

**Review by iamveritas — APPROVED** [review]

**Pull request merged**